### PR TITLE
chore: Reorganise IDE Panels

### LINKED ISSUE #31387: [Bug]: Entering or exiting split pane mode with query selected - query's title gets selected for renaming
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Description

When I open up or close out the split pane view with a query selected, the query's title gets selected for edit/renaming

### Steps To Reproduce

https://github.com/appsmithorg/appsmith/assets/101863839/b1c7487f-045d-4522-854e-4217a2865cbc

### Public Sample App

_No response_

### Environment

Release

### Issue video log

_No response_

### Version

Cloud

### LINKED ISSUE #31707: [Bug]: Transition between edit and preview modes appear glitchy, more noticeable when modal is open
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Description

Transition between edit and preview modes appear glitchy when a modal widget is open

### Steps To Reproduce

1. Toggle between view and preview mode

### Public Sample App

_No response_

### Environment

Production

### Severity

Medium (Frustrating UX)

### Issue video log

https://www.loom.com/share/060821de64434fb493f91927f281d2af

### Version

Cloud

### LINKED ISSUE #34545: [Task]: Render only Data and Canvas on the main pane

### PR DESCRIPTION
**Pull request by hetunandu**
## Description

By re organising the panels in the IDE, we are creating a new Code Editor only panel which is separate from the main / canvas and the explorer panel. This ensures that the Code Editor is not remounted when switching between Split screen and Full screen. This PR is also improving the Preview mode by not changing urls when entering and exiting preview modes.

EE PR: https://github.com/appsmithorg/appsmith-ee/pull/4760

Fixes #31707
Fixes #31387
Fixes #34545
Fixes #34546

## Automation

/ok-to-test tags="@tag.All"

### :mag: Cypress test results

> [!TIP]
> 🟢 🟢 🟢 All cypress tests have passed! 🎉 🎉 🎉
> Workflow run: <https://github.com/appsmithorg/appsmith/actions/runs/10177085462>
> Commit: 26688b0fde7e797d88996ef569d9fc06e3942958
> <a href="https://internal.appsmith.com/app/cypress-dashboard/rundetails-65890b3c81d7400d08fa9ee5?branch=master&workflowId=10177085462&attempt=1" target="_blank">Cypress dashboard</a>.
> Tags: `@tag.All`
> Spec:
> <hr>Wed, 31 Jul 2024 09:42:50 UTC

## Communication
Should the DevRel and Marketing teams inform users about this change?
- [ ] Yes
- [ ] No

## Summary by CodeRabbit

- **New Features**
	- Introduced `EditorPaneExplorer` and `QueryExplorer` components for enhanced navigation and query management within the IDE.
	- Added visual enhancements to `ListJSObjects` and `AddWidgets` components with new style properties.

- **Improvements**
	- Implemented clamping in the `useDynamicAppLayout` hook to prevent negative width values, enhancing layout stability.
	- Updated layout management to ensure a responsive user experience based on IDE view modes.

- **Style**
	- Enhanced visual clarity by adding borders to various components, improving UI separation and organization.

- **Bug Fixes**
	- Streamlined test cases in the `Postgres2_Spec.ts` to maintain coherent sequencing after removing unnecessary tests.

### CODE REVIEW

File: app/client/src/pages/Editor/IDE/EditorTabs/List.tsx [hunk 1]
```diff
@@ -24,9 +24,9 @@ export const List = () => {
   return (
     <ListContainer
       bg="var(--ads-v2-color-bg)"
-      className="absolute top-[78px]"
+      className="absolute top-[32px]"
```
Comment (albinAppsmith) [inline app/client/src/pages/Editor/IDE/EditorTabs/List.tsx:27]: nit: Should we move this to constants? Or can you add a comment on why is 32?

**Comment by albinAppsmith** [comment]
@hetunandu DO we have EE PR?

**Comment by albinAppsmith** [comment]
New Query page UI seems broken
Screenshot 2024-07-25 at 10 36 41

**Review by alex-golovanov — APPROVED** [review]

**Pull request merged**